indexer/packages/postgres/src/stores/pnl-ticks-table.ts

```diff
@@ -268,3 +290,159 @@ export async function findMostRecentPnlTickForEachAccount(
+      WHERE
+          "createdAt"::date = CURRENT_DATE
+          AND (b."subaccountNumber" % 128) = 0
+          AND b."address" NOT IN (${vaultAddressesString})
+    ), latest_pnl as(
+      SELECT
+          "subaccountId",
+          "totalPnl",
+          "netTransfers",
+          "currentEquity" ,
+          "address"
+      FROM 
+          latest_pnl_ranked
+      WHERE
+          "rn" = 1
+    ), subaccount_pnl_difference as(
+      SELECT
+        a."address",
+        a."totalPnl" - COALESCE(b."totalPnl", 0) as "pnlDifference",
+        a."netTransfers" as "netTransfers",
+        a."currentEquity" as "currentEquity"
+      FROM latest_pnl a left join latest_subaccount_pnl_x_days_ago b
+      ON a."subaccountId"=b."subaccountId"
+    ), aggregated_results as(
+    SELECT
+      "address",
+      sum(subaccount_pnl_difference."pnlDifference") as "totalPnl",
+      sum(subaccount_pnl_difference."netTransfers") as "netTransfers",
+      sum(subaccount_pnl_difference."currentEquity") as "currentEquity"
+    FROM
+      subaccount_pnl_difference
+    GROUP BY address
+    )
+    SELECT
+      "address",
+      "totalPnl" as "pnl",
+      '${timeSpan}' as "timeSpan",
+      "currentEquity",
+      ROW_NUMBER() over (order by aggregated_results."totalPnl" desc) as rank
+    FROM
+      aggregated_results;
+    `,
+  ) as { rows: LeaderboardPnlCreateObject[] };
+
+  return result.rows;
+}
+
+async function getLatestRankedPnlTicks(): Promise<LeaderboardPnlCreateObject[]> {
+  const vaultAddresses: string[] = getVaultAddresses();
+  const vaultAddressesString: string = vaultAddresses.map((address) => `'${address}'`).join(',');
+  const result: {
+    rows: LeaderboardPnlCreateObject[]
+  } = await knexReadReplica.getConnection().raw(
+    `
+    WITH latest_pnl_ranked as (
+      SELECT
+          "subaccountId",
+          "totalPnl",
+          "netTransfers",
+          "equity" as "currentEquity",
+          "address",
+          ROW_NUMBER() OVER (PARTITION BY "subaccountId" ORDER BY "blockHeight" DESC) AS "rn"
+      FROM
+          pnl_ticks a left join subaccounts b ON a."subaccountId"=b."id"
+      WHERE
+          "createdAt"::date = CURRENT_DATE
+          AND (b."subaccountNumber" % 128) = 0
+          AND b."address" NOT IN (${vaultAddressesString})
+    ), latest_pnl as(
+      SELECT
+          "subaccountId",
+          "totalPnl",
+          "netTransfers",
+          "currentEquity" ,
+          "address"
+      FROM 
+          latest_pnl_ranked
+      WHERE
+          "rn" = 1
+    ), aggregated_results as(
+    SELECT
+      "address",
+      sum(latest_pnl."totalPnl") as "totalPnl",
+      sum(latest_pnl."netTransfers") as "netTransfers",
+      sum(latest_pnl."currentEquity") as "currentEquity"
+    FROM
+      latest_pnl
+    GROUP BY address
+    )
+    SELECT
+      "address",
+      "totalPnl" as "pnl",
+      'ALL_TIME' as "timeSpan",
+      "currentEquity",
```

Comment:
thinking about this a little more, if we want to calculate pnl over this period of time by percentage isn't it:

`(ending equity - starting equity - net transfers)/(starting equity + netTransfers/2)`

Let's check one more time with Yi that we don't want percentage PNL

Reply:
Why netTransfers/2?
old way of doing this was `Percent PNL = (Dollar PNL) / max(500, Starting Equity + Deposits During Period)`

Reply:
Added linear ticket for this one https://linear.app/dydx/project/social-mvp-e029a2c846e4/issues. Since I need to change tables etc

Reply:
we can stick to this formula if thats what we've used historically. I do think we should use net deposits rather than deposits during period though